src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -154,6 +161,11 @@ public SemanticModel(Compilation compilation, BicepFile sourceFile, IFileResolve
         /// </summary>
         public ImmutableArray<DeclaredResourceMetadata> DeclaredResources => declaredResourcesLazy.Value;
 
+        /// <summary>
+        /// Gets function variables that are emitted by function calls. Used mostly to optimise loading content from files.
+        /// </summary>
+        public IImmutableDictionary<FunctionCallSyntaxBase, FunctionVariable> FunctionVariables => functionVariablesLazy.Value;
```

Comment:
Could we move this to the `EmitterContext` class, since this is an optmization for code gen?

Reply:
Good point, will move it.

Reply:
Moved.